sql/core/src/main/scala/org/apache/spark/sql/streaming/ContinuousQueryListener.scala

```diff
@@ -70,26 +71,34 @@ abstract class ContinuousQueryListener {
 object ContinuousQueryListener {
 
   /**
-   * Base type of [[ContinuousQueryListener]] events.
+   * Base type of [[ContinuousQueryListener]] events
    * @since 2.0.0
    */
-  trait Event
+  trait Event extends SparkListenerEvent
 
   /**
-   * Event representing the start of a query.
+   * Event representing the start of a query
    * @since 2.0.0
    */
-  class QueryStarted private[sql](val query: ContinuousQuery) extends Event
+  class QueryStarted private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /**
-   * Event representing any progress updates in a query.
+   * Event representing any progress updates in a query
    * @since 2.0.0
    */
-  class QueryProgress private[sql](val query: ContinuousQuery) extends Event
+  class QueryProgress private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /**
-   * Event representing that termination of a query.
+   * Event representing that termination of a query
+   *
+   * @param queryInfo The query info.
+   * @param exception The exception message of the [[ContinuousQuery]] if any. Otherwise, it
+   *                  will be `None`.
+   * @param stackTrace The stack trace of the exception if any.
    * @since 2.0.0
    */
-  class QueryTerminated private[sql](val query: ContinuousQuery) extends Event
+  class QueryTerminated private[sql](
+      val queryInfo: ContinuousQueryInfo,
+      val exception: Option[String],
+      val stackTrace: Seq[StackTraceElement]) extends Event
```

Comment:
Isnt it more Java friendly to make this Array instead of Seq? It does not need to be Seq.